### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by HermesSbicego-Laser** [OrchardCMS/Orchard#8354]
We are experiencing performance issues using admin panel with more than 2 users.
SQL profiler tells that `select shelldescr0_.Id as Id38_, shelldescr0_.SerialNumber as SerialNu2_38_ from Settings_ShellDescriptorRecord shelldescr0_ where shelldescr0_.Id is not null` is fired hundreds of times.
We have a fix ready for  that, but we will go deeper in order to find a way to have orchard performing better.

**Comment by MatteoPiovanelli** [comment]
Expanding on this, even with the improvements from #8355, the database was being hit for the lists of ShellFeatureRecord and ShellParameterRecord on every request. Running sql traces, those queries were noticeable. So we wrote and tested a version of ShellDescriptorManager that caches all that. This will target dev rather than 1.10.x because we used ILockingProvider for an application lock around the part of code that handles updating the records and evicting the cache.
